File: app/src/main/java/com/cargopull/executor_driver/gateway/OrderCostDetailsFilter.java
package com.cargopull.executor_driver.gateway;

import io.reactivex.functions.Predicate;
import ua.naiksoftware.stomp.client.StompMessage;

public class OrderCostDetailsFilter implements Predicate<StompMessage> {

  @Override
  public boolean test(StompMessage stompMessage) {
    return "PAYMENT_CONFIRMATION".equals(stompMessage.findHeader("Status"));
  }
}

File: app/src/main/java/com/cargopull/executor_driver/gateway/OrderCostDetailsFilter.kt
package com.cargopull.executor_driver.gateway

import com.cargopull.executor_driver.backend.stomp.StompFrame
import io.reactivex.functions.Predicate

class OrderCostDetailsFilter : Predicate<StompFrame> {

    override fun test(stompFrame: StompFrame): Boolean {
        return "PAYMENT_CONFIRMATION" == stompFrame.headers.get("Status")
    }
}
